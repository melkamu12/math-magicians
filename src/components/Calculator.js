import React from 'react';
import ButtonCalculator from './ButtonCalc';

const Calculator = () => {
  return (
    <div className="Math-Calculator">
      <div className="result-display">
        <span className="result">0</span>
      </div>
      <ButtonCalculator />
    </div>
  );
};

export default Calculator;
