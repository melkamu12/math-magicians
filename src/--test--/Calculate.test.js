import React from 'react';
import { render } from '@testing-library/react';
import Calculator from '../components/Calculator';
import '@testing-library/jest-dom/extend-expect';
import calculate from '../logic/calculate';

describe('Calculator component', () => {
  it('should match the snapshot', () => {
    const { container } = render(<Calculator />);
    expect(container).toMatchSnapshot();
  });
  test('clears the calculator data when AC button is pressed', () => {
    const result = calculate({ total: 10, next: 5, operation: '+' }, 'AC');
    expect(result).toEqual({ total: 0, next: null, operation: null });
  });

  test('clears the calculator data when AC button is pressed', () => {
    const result = calculate({ total: 10, next: 5, operation: '+' }, 'AC');
    expect(result).toEqual({ total: 0, next: null, operation: null });
  });

  test('handles decimal point correctly when "." button is pressed', () => {
    const result = calculate({ total: 10, next: '5', operation: null }, '.');
    expect(result).toEqual({ total: 10, next: '5.', operation: null });
  });

  test('performs subtraction correctly when "-" button is pressed', () => {
    const result = calculate({ total: 10, next: '5', operation: null }, '-');
    expect(result).toEqual({ total: '5', next: null, operation: '-' });
  });

  test('handles division by zero gracefully', () => {
    const result = calculate({ total: 10, next: '0', operation: '/' }, '=');
    expect(result).toEqual({
      total: "Can't divide by 0.",
      next: null,
      operation: null,
    });
  });

  test('changes the sign of the number when "+/-" button is pressed', () => {
    const result = calculate({ total: 10, next: '5', operation: null }, '+/-');
    expect(result).toEqual({ total: 10, next: '-5', operation: null });
  });

  test('handles operation chaining correctly', () => {
    const result = calculate({ total: 10, next: '5', operation: '+' }, '-');
    expect(result).toEqual({ total: '15', next: null, operation: '-' });
  });
});
