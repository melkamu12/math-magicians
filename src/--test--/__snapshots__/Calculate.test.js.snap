// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component should match the snapshot 1`] = `
<div>
  <div
    class="Math-Calculator"
  >
    <div
      class="result-display"
    >
      <span
        class="result"
      >
        0
      </span>
    </div>
    <div
      class="keyButtons"
    >
      <button
        class="btn"
        data="AC"
        type="button"
      >
        AC
      </button>
      <button
        class="btn"
        data="+/-"
        type="button"
      >
        +/-
      </button>
      <button
        class="btn"
        data="%"
        type="button"
      >
        %
      </button>
      <button
        class="btn-operator"
        data="/"
        type="button"
      >
        ÷
      </button>
      <button
        class="btn"
        data="7"
        type="button"
      >
        7
      </button>
      <button
        class="btn"
        data="8"
        type="button"
      >
        8
      </button>
      <button
        class="btn"
        data="9"
        type="button"
      >
        9
      </button>
      <button
        class="btn-operator"
        data="*"
        type="button"
      >
        ×
      </button>
      <button
        class="btn"
        data="4"
        type="button"
      >
        4
      </button>
      <button
        class="btn"
        data="5"
        type="button"
      >
        5
      </button>
      <button
        class="btn"
        data="6"
        type="button"
      >
        6
      </button>
      <button
        class="btn-operator"
        data="-"
        type="button"
      >
        –
      </button>
      <button
        class="btn"
        data="1"
        type="button"
      >
        1
      </button>
      <button
        class="btn"
        data="2"
        type="button"
      >
        2
      </button>
      <button
        class="btn"
        data="3"
        type="button"
      >
        3
      </button>
      <button
        class="btn-operator"
        data="+"
        type="button"
      >
        +
      </button>
      <button
        class="btn btn-key"
        data="0"
        type="button"
      >
        0
      </button>
      <button
        class="btn"
        data="."
        type="button"
      >
        .
      </button>
      <button
        class="btn-operator"
        data="="
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
